frontend/src/containers/pages/AuthoringTool/CanvasSideBar/ComponentProperties/ButtonPropertiesComponent.jsx

```diff
@@ -25,6 +32,47 @@ export default function ButtonPropertiesComponent({
 }) {
   const { scenes, updateComponentProperty } = useContext(SceneContext);
 
+  const [newFlag, setNewFlag] = useState("");
+  const [open, setOpen] = useState(false);
+
+  const handleInputChange = (event) => {
+    setNewFlag(event.target.value);
+  };
+
+  const handleAddFlag = () => {
+    if (Object.keys(component.flagList).includes(`add ${newFlag}`)) {
```

Comment:
Potential `undefined` accessor -  on legacy scenarios, flagList may not exist.

Reply:
Oop, I'll change that.